### PR TITLE
docs: developing ica auth module

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

ref: #512 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/app_modules/interchain-accounts/ica_auth.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+<!--
+order: 3
+-->
+
+# Building an ICA Authentication Module
```
Comment (damiannolan) [inline docs/app_modules/interchain-accounts/ica_auth.md:5]: Should we maybe link to or reference ICS-30 in this intro section?

File: docs/app_modules/interchain-accounts/ica_auth.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+The controller module is used for account registration and packet sending. 
+It executes only logic required of all controllers of interchain accounts. 
+The type of authentication used to manage the interchain accounts remains unspecified. 
+There may exist may different types of authentication which are desirable for different use cases.
```
Comment (damiannolan) [inline docs/app_modules/interchain-accounts/ica_auth.md:10]:
```suggestion
There may exist many different types of authentication which are desirable for different use cases. 
```

File: docs/app_modules/interchain-accounts/ica_auth.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+The type of authentication used to manage the interchain accounts remains unspecified. 
+There may exist may different types of authentication which are desirable for different use cases. 
+Thus the purpose of the authentication module is to wrap the controller module with custom authentication logic.
+
+In ibc-go, authentication modules are connected to the controller chain via a middleware stack. 
+The authentication module acts as the base application within this stack. 
+To implement an authentication module, the `IBCModule` interface must be fulfilled. 
+By implementing the controller module as middleware, any amount of authentication modules can be created and connected to the controller module without writing redundant code. 
+
+The authentication module must:
```
Comment (damiannolan) [inline docs/app_modules/interchain-accounts/ica_auth.md:18]: Should we add claiming of channel capability to this bulleted list? I know it is mentioned below but just a thought as I read through.

But maybe it's okay to omit(?) if this intended to be a higher level overview. Your call!

**Review by damiannolan — APPROVED** [review]
Nice work 👍 
I think it may be useful to reference ICS-30 or link to the middleware docs somewhere at the beginning of the doc

**Review by crodriguezvega — CHANGES_REQUESTED** [review]
Thanks for writing his docs, @colin-axner. Sorry that I got a bit too excited with the alignments... I think you should check them in the whole document, there are other places where they are off that I didn't comment on.

File: docs/app_modules/interchain-accounts/ica_auth.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+	Data: data,
+}
+
+_, err = keeper.icaControllerKeeper.TrySendTx(ctx, chanCap, p, packetData)
+```
+
+The data within an `InterchainAccountPacketData` must be serialized using a format supported by the host chain. 
+If the host chain is using the ibc-go host chain submodule, `SerializeCosmosTx` should be used. If the `InterchainAccountPacketData.Data` is serialized using a format not support by the host chain, the packet will not be successfully received.  
+
+### Integration into app.go file
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply docs/app_modules/interchain-accounts/ica_auth.md:217]: Agreed, which format do we want to go with. 

Also in this document I only see the main title as having multiple words capitalized, but "Authentication Module" could arguably capitalized as the name of something. Should I change this?

**Review by seantking — APPROVED** [review]

File: docs/app_modules/interchain-accounts/ica_auth.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply docs/app_modules/interchain-accounts/ica_auth.md:219]: I would say let's go with capitalizing only the first word. And wrt to authentication module, I don't see a reason to capitalize it, so I would also leave it uncapitalized.

File: docs/app_modules/interchain-accounts/ica_auth.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
```
Comment (crodriguezvega) [inline docs/app_modules/interchain-accounts/ica_auth.md:219]:
```suggestion
### Integration into `app.go` file
```

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**